addons/account/models/account_reconcile_model.py

```diff
@@ -746,23 +758,32 @@ def _apply_rules(self, st_lines, excluded_ids=None, partner_map=None):
                     first_batch_candidates_proposed = []
                     second_batch_candidates = []
                     second_batch_candidates_proposed = []
+                    third_batch_candidates = []
+                    third_batch_candidates_proposed = []
                     for c in candidates:
                         # Don't take into account already reconciled lines.
                         if c['aml_id'] in reconciled_amls_ids:
                             continue
 
                         # Dispatch candidates between lines matching invoices with the communication or only the partner.
-                        elif c['communication_flag']:
+                        elif c['payment_reference_flag']:
                             if c['aml_id'] in amls_ids_to_exclude:
                                 first_batch_candidates_proposed.append(c)
                             else:
                                 first_batch_candidates.append(c)
-                        elif not first_batch_candidates:
+                        elif c['communication_flag']:
                             if c['aml_id'] in amls_ids_to_exclude:
                                 second_batch_candidates_proposed.append(c)
                             else:
                                 second_batch_candidates.append(c)
-                    available_candidates = first_batch_candidates + first_batch_candidates_proposed or second_batch_candidates + second_batch_candidates_proposed
+                        elif not first_batch_candidates:
```

Comment:
`... and not second_batch_candidates` ? I'm not sure

Reply:
I _think_ its good now